### DISCUSSION TITLE
An IImageWebProcessor that always runs

### DISCUSSION THREAD

**Discussion by cosullivan** [SixLabors/ImageSharp.Web#133] [Ideas]
Hi,

We have the requirement to do some default processing on all images regardless of whether any commands have been specified. At the moment the processors will only get executed when there is at least one command that has been specified. 

The only way I can see to achieve this at the moment is via the OnProcessedAsync callback but then I would have to read and write the image again.

If the IWebImageProcessor included something like a CanProcess(...) method that was called then the processors could the determine for themselves whether they can operate on the image as opposed to only operating if their command has been specified. 

I am happy to help out with the development if required.

Thanks, Cain.

**Comment by JimBobSquarePants** [comment]
Sorry @cosullivan I just saw this. 

`IWebImageProcessor` implementations iterated after commands have been parsed. Adding that property won't help.

You have two options.

Use `ImageSharpMiddlewareOptionsh.OnParseCommandsAsync` to insert your own command to then be picked up by your processor. 

https://github.com/SixLabors/ImageSharp.Web/blob/b72064b3b8cb8b883f8310c86b6d7e5643d80ad3/src/ImageSharp.Web/Middleware/ImageSharpMiddlewareOptions.cs#L97

or

Inherit the `PhysicalFileSystemProvider` and override the `ProcessingBehavior` property.

https://github.com/SixLabors/ImageSharp.Web/blob/b72064b3b8cb8b883f8310c86b6d7e5643d80ad3/src/ImageSharp.Web/Providers/PhysicalFileSystemProvider.cs#L50

https://github.com/SixLabors/ImageSharp.Web/blob/b72064b3b8cb8b883f8310c86b6d7e5643d80ad3/src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs#L221
